src/Val.cc

```diff
@@ -806,13 +729,14 @@ uint32_t PortVal::Mask(uint32_t port_num, TransportProto port_type)
 	return port_num;
 	}
 
-PortVal::PortVal(uint32_t p) : Val(bro_uint_t(p), TYPE_PORT)
+PortVal::PortVal(uint32_t p)
+	: UnsignedValImplementation(base_type(TYPE_PORT), bro_uint_t(p))
 	{
 	}
 
 uint32_t PortVal::Port() const
 	{
-	uint32_t p = static_cast<uint32_t>(val.uint_val);
+	uint32_t p = static_cast<uint32_t>(uint_val);
```

Comment:
I don't know the degree to which those typedefs are consistently used, but the if they were, then an answer is that it potentially adds clarity: it's "the width of the scripting language's integer values" versus "this code uses a 64-bit integer because that's what we think is required here for whatever reason".   E.g. the hypothetical act of looking for locations in the code that use "scripting language integer values" is easier if that concept has a distinct type name (not that I think that's frequent or expected to change again, but for historical reference, the typedef was originally to a 32-bit integer, not 64-bit, so such an activity isn't beyond imagination).

Reply:
Make sense. Thanks for the explanation 👍